drgn/helpers/linux/net.py

```diff
@@ -45,3 +43,25 @@ def sk_nulls_for_each(head: Object) -> Iterator[Object]:
         "struct sock", head, "__sk_common.skc_nulls_node"
     ):
         yield sk
+
+
+_NETDEV_HASHBITS = 8
+_NETDEV_HASHENTRIES = 1 << _NETDEV_HASHBITS
+
+
+def find_netdev_by_index(net: Object, ifindex: int) -> Object:
+    """
+    Return the ``struct net_device`` object of the given interface index
+    number, or ``NULL`` if not found.
+
+    :param net: ``struct net *``
```

Comment:
We should also allow the first argument to be `prog` and automatically translate that to `init_net`. See [`path_lookup`](https://github.com/osandov/drgn/blob/51f63bb53b2fe8adb5b685f9215773aaaf7fb55b/drgn/helpers/linux/fs.py#L102).